force-app/main/default/classes/TaskTriggerHandler.cls

```diff
@@ -1,53 +1,47 @@
    
     public static void checkOnCompleteTask(List<Task> taskList){
-        List<Contact> toUpdate = [Select Id, Name, Points__c,
-                                  	(
-                                    	Select Id,ActivityDate,CreatedDate ,Task_Type__c,Points__c,IsClosed
-                                        FROM Tasks
-                                        WHERE Id IN :taskList
-                                        AND IsClosed = true
-                                    )	
-                                  FROM Contact];
+        List<Contact> selectedContacts = [Select Id, Name, Points__c,
+                                        (
+                                            Select Id,ActivityDate,CreatedDate ,Task_Type__c,Points__c,IsClosed
+                                            FROM Tasks
+                                            WHERE Id IN :taskList
+                                            AND IsClosed = true
+                                        )	
+                                      	FROM Contact];
         
-        for(Contact c : toUpdate)
+		List<Contact> toUpdate = new List<Contact>();
+        
+        for(Contact c : selectedContacts)
         {
             if(c.tasks != null)
             {
                 double contactPoints = c.Points__c == null ? 0 : c.Points__c;
                 
                 for(Task task : c.tasks)
                 {
-                    if(task.IsClosed)
-                    {
-                        if(task.Task_Type__c == 'TO-DO')
-                        {
-                            if(task.ActivityDate != null)
-                                {
-                                   if(task.ActivityDate < Date.today())
-                                   {
-                                       contactPoints += task.Points__c;
-                                   }else if(task.ActivityDate >= Date.today())
-                                   {
-                                       contactPoints -= task.Points__c / 2;
-                                   }
-                                }else {
-                                    contactPoints += task.Points__c;
-                                }                      
-                        }else  if(task.Task_Type__c == 'Daily')
-                        {
-                            if(task.CreatedDate < Date.today())
-                            {
-                                contactPoints -= task.Points__c;
-                            }else{
-                                contactPoints += task.Points__c;
-                            }
-                        }
-                    }
+                    contactPoints += processTaskPoints(task);
                 }
                 c.Points__c = contactPoints;
+                toUpdate.add(c);
             }
         }      
-        update toUpdate;
+        if(toUpdate.size() >= 1)
+        {
+             update toUpdate;
```

Comment:
if you improve the name of the variable, then you wont have "update toUpdate" hehe